rmw_connext_shared_cpp/src/shared_functions.cpp

```diff
@@ -346,6 +347,103 @@ create_node(
     return NULL;
   }
 
+  if (security_options->security_root_path) {
```

Comment:
You must check that `security_options` is not null before accessing its members.